library/general/src/lib/ui/multi_messages_dialog.rb

```diff
@@ -0,0 +1,247 @@
+require "yast"
+require "ui/dialog"
+
+Yast.import "UI"
+Yast.import "Label"
+
+module UI
+  # This dialog receives several messages and allows the user to browse them
+  # using 'Next' and 'Back' buttons.
+  #
+  # Each message is an object that should respond to #title and #body methods.
+  class MultiMessagesDialog < ::UI::Dialog
+    include Yast::Logger
+
+    DEFAULT_MIN_HEIGHT = 20
+    DEFAULT_MIN_WIDTH = 60
+    TIMEOUT_STEP = 1000 # miliseconds
+
+    attr_reader :messages, :headline, :min_height, :min_width, :timeout
+
+    # Constructor
+    #
+    # @param headline   [String]  Dialog's headline
+    # @param messages   [Array]   Array of messages. Objects in the array should
+    #                             respond to #title and #body methods.
```

Comment:
NP: Maybe this file should provide a generic message class? I'm not sure if we want to create a separate class at every place where this dialog is used...